File: packages/ui-compose/ui/src/test/kotlin/com/kinetixui/ui/blocks/SettingsListBlockTest.kt
package com.kinetixui.ui.blocks

import androidx.compose.foundation.layout.width
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Modifier
import androidx.compose.ui.test.junit4.v2.createComposeRule
import androidx.compose.ui.test.onNodeWithText
import androidx.compose.ui.unit.dp
import com.kinetixui.ui.KinetixCard
import com.kinetixui.ui.KinetixCardHeader
import com.kinetixui.ui.KinetixCardTitle
import com.kinetixui.ui.KinetixList
import com.kinetixui.ui.KinetixListItem
import com.kinetixui.ui.KinetixSeparator
import com.kinetixui.ui.KinetixSwitch
import com.kinetixui.ui.KinetixTheme
import org.junit.Rule
import org.junit.Test
import org.junit.runner.RunWith
import org.robolectric.RobolectricTestRunner
import org.robolectric.annotation.Config

// The "Settings list" block — see TestimonialBlockTest.kt for how block fixtures work.
//
// kx-block:start
@Composable
fun SettingsListBlock() {
    var email by remember { mutableStateOf(true) }
    var push by remember { mutableStateOf(true) }
    var sms by remember { mutableStateOf(false) }

    KinetixCard(Modifier.width(360.dp)) {
        KinetixCardHeader { KinetixCardTitle("Notifications") }
        KinetixSeparator()
        KinetixList {
            KinetixListItem(
                title = "Email",
                description = "Product news and receipts",
                trailing = { KinetixSwitch(email, { email = it }) },
            )
            KinetixListItem(
                title = "Push",
                description = "Activity on your projects",
                trailing = { KinetixSwitch(push, { push = it }) },
            )
            KinetixListItem(
                title = "SMS",
                description = "Only critical alerts",
                trailing = { KinetixSwitch(sms, { sms = it }) },
            )
        }
    }
}
// kx-block:end

@RunWith(RobolectricTestRunner::class)
@Config(sdk = [34])
class SettingsListBlockTest {
    @get:Rule
    val rule = createComposeRule()

    @Test
    fun renders_every_row_with_its_description() {
        rule.setContent { KinetixTheme(darkTheme = false) { SettingsListBlock() } }
        rule.onNodeWithText("Notifications").assertExists()
        rule.onNodeWithText("Email").assertExists()
        rule.onNodeWithText("Only critical alerts").assertExists()
    }
}

File: packages/ui-flutter/test/blocks/settings_list_block_test.dart
import 'package:flutter/material.dart';
import 'package:flutter_test/flutter_test.dart';
import 'package:kinetix_ui/kinetix_ui.dart';

// The "Settings list" block — see testimonial_block_test.dart for how block fixtures work.
//
// kx-block:start
class SettingsListBlock extends StatefulWidget {
  const SettingsListBlock({super.key});

  @override
  State<SettingsListBlock> createState() => _SettingsListBlockState();
}

class _SettingsListBlockState extends State<SettingsListBlock> {
  bool _email = true;
  bool _push = true;
  bool _sms = false;

  @override
  Widget build(BuildContext context) {
    return KinetixCard(
      child: Column(
        mainAxisSize: MainAxisSize.min,
        children: <Widget>[
          const KinetixCardHeader(children: <Widget>[KinetixCardTitle('Notifications')]),
          const KinetixSeparator(),
          KinetixList(
            children: <Widget>[
              KinetixListItem(
                title: 'Email',
                description: 'Product news and receipts',
                trailing: KinetixSwitch(value: _email, onChanged: (bool v) => setState(() => _email = v)),
              ),
              KinetixListItem(
                title: 'Push',
                description: 'Activity on your projects',
                trailing: KinetixSwitch(value: _push, onChanged: (bool v) => setState(() => _push = v)),
              ),
              KinetixListItem(
                title: 'SMS',
                description: 'Only critical alerts',
                trailing: KinetixSwitch(value: _sms, onChanged: (bool v) => setState(() => _sms = v)),
              ),
            ],
          ),
        ],
      ),
    );
  }
}
// kx-block:end

void main() {
  testWidgets('renders every row with its description', (WidgetTester tester) async {
    await tester.pumpWidget(
      MaterialApp(
        home: KinetixTheme(
          brightness: Brightness.light,
          child: const Scaffold(body: SingleChildScrollView(child: SettingsListBlock())),
        ),
      ),
    );

    expect(find.text('Notifications'), findsOneWidget);
    expect(find.text('Email'), findsOneWidget);
    expect(find.text('Only critical alerts'), findsOneWidget);
  });
}
